File: largest-number-at-least-twice-of-others/largest-number-at-least-twice-of-others.java
class Solution {
    public int dominantIndex(int[] nums) {
        int maxValue = -1, maxIndex = -1, secondMax = -1;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] > maxValue) {
                secondMax = maxValue;
                maxValue = nums[i];
                maxIndex = i;
            } else if (nums[i] > secondMax)
                secondMax = nums[i];
        }
        return maxValue < secondMax * 2 ? -1 : maxIndex;
    }
}

File: largest-number-at-least-twice-of-others/largest-number-at-least-twice-of-others.kt
class Solution {
    fun dominantIndex(nums: IntArray): Int {
        var maxValue = -1
        var maxIndex = -1
        nums.forEachIndexed { index, value ->
            if (value > maxValue) {
                maxValue = value
                maxIndex = index
            }
        }
        nums.forEach {
            if (it != maxValue && maxValue < it * 2)
                return -1
        }
        return maxIndex
    }
}